modules/apps/29-fee/keeper/relay.go

```diff
@@ -0,0 +1,34 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/modules/core/exported"
+)
+
+// SendPacket wraps IBC ChannelKeeper's SendPacket function
+func (k Keeper) SendPacket(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI) error {
+	return k.ics4Wrapper.SendPacket(ctx, chanCap, packet)
+}
+
+// WriteAcknowledgement wraps IBC ChannelKeeper's WriteAcknowledgement function
+// ICS29 WriteAcknowledgement is used for asynchronous acknowledgements
+func (k Keeper) WriteAcknowledgement(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI, acknowledgement []byte) error {
```

Comment:
@AdityaSripal I guess I need the `OnRecvPacket` functionality finished before I can write a test for this? 

#262

Reply:
Yea because `OnRecvPacket` should ONLY set `ForwardRelayerAddress` if the underlying application returns a `nil` acknowledgement. cc: @charleenfei 

No need to store it if you're going to send it out immediately